crates/fsrt/src/main.rs

```diff
@@ -653,10 +687,21 @@ pub(crate) fn scan_directory<'a>(
         used_graphql_perms.extend_from_slice(&graphql_perms_defid);
         used_graphql_perms.extend_from_slice(&graphql_perms_varid);
 
-        let final_perms: Vec<&String> = perm_interp
+        let oauth_scopes: HashSet<&&String> = used_graphql_perms
+            .iter()
+            .filter_map(|val| {
+                if !scope_name_to_oauth.contains_key(&val) {
+                    warn!("Scope is not contained in the scope definitions")
+                }
+
+                scope_name_to_oauth.get(&val)
```

Comment:
You can deref the result to get a `HashSet<&str>`.

Reply:
Fixed. Thank you for the help!